sklearnex/basic_statistics/incremental_basic_statistics.py

```diff
@@ -0,0 +1,276 @@
+            Mean of each feature over all samples.
+
+        variance : ndarray of shape (n_features,)
+            Variance of each feature over all samples.
+
+        variation : ndarray of shape (n_features,)
+            Variation of each feature over all samples.
+
+        sum_squares : ndarray of shape (n_features,)
+            Sum of squares for each feature over all samples.
+
+        standard_deviation : ndarray of shape (n_features,)
+            Standard deviation of each feature over all samples.
+
+        sum_squares_centered : ndarray of shape (n_features,)
+            Centered sum of squares for each feature over all samples.
+
+        second_order_raw_moment : ndarray of shape (n_features,)
+            Second order moment of each feature over all samples.
+    """
+
+    _onedal_incremental_basic_statistics = staticmethod(onedal_IncrementalBasicStatistics)
+
+    if sklearn_check_version("1.2"):
+        _parameter_constraints: dict = {
+            "result_options": [
+                StrOptions(
+                    {
+                        "all",
+                        "min",
+                        "max",
+                        "sum",
+                        "mean",
+                        "variance",
+                        "variation",
+                        "sum_squares",
+                        "standard_deviation",
+                        "sum_squares_centered",
+                        "second_order_raw_moment",
+                    }
+                ),
+                list,
+            ],
+            "batch_size": [Interval(numbers.Integral, 1, None, closed="left"), None],
+        }
+
+    def __init__(self, result_options="all", batch_size=None):
+        if result_options == "all":
+            self.result_options = (
+                self._onedal_incremental_basic_statistics.get_all_result_options()
+            )
+        else:
+            self.result_options = result_options
+        self._need_to_finalize = False
+        self.batch_size = batch_size
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearn.covariance.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
+
+    _onedal_cpu_supported = _onedal_supported
+    _onedal_gpu_supported = _onedal_supported
+
+    def _get_onedal_result_options(self, options):
+        if isinstance(options, list):
+            onedal_options = "|".join(self.result_options)
+        else:
+            onedal_options = options
+        assert isinstance(onedal_options, str)
+        return options
+
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        self._onedal_estimator.finalize_fit()
+        self._need_to_finalize = False
+
+    def _onedal_partial_fit(self, X, weights, queue):
+        first_pass = not hasattr(self, "n_samples_seen_") or self.n_samples_seen_ == 0
```

Comment:
are first_pass and n_samples_seen_ needed for incremental basic stats?

Reply:
first_pass is used below. and n_samples_seen_ is added for conformance. I admit there can be scenarios where user needs it.